tree/tree.go

```diff
@@ -121,6 +121,24 @@ func (t *Tree) Add(wTx db.WriteTx, key, value []byte) error {
 	return nil
 }
 
+// Del deletes a key from the Tree.
```

Comment:
What about: 

```
// Del deletes a key from the Tree.
func (t *Tree) Del(wTx db.WriteTx, key []byte) error {
	givenTx := wTx == nil
	if givenTx {
		wTx = t.DB().WriteTx()
		defer wTx.Discard()
	}
	if err := wTx.Delete(key); err != nil {
		return err
	}
	if givenTx {
		return wTx.Commit()
	}
	return nil
}
```
- If the negation operator is removed the code is more readable (IMO)
- In the last condition check we can just return the `wtx.Commit()` so there is no need to check the error